### DISCUSSION TITLE
I want a feature to be added to PlotJuggler

### DISCUSSION THREAD

**Discussion by facontidavide** [facontidavide/PlotJuggler#520] [General]
![image](https://user-images.githubusercontent.com/2822888/133937881-069f69cc-4ba7-4498-94a4-50c91ac30cf2.png)

Hi,

first of all, let me say that it is great that users interact with me and suggest new features. I received a lot of great ideas from the community and they helped to make PlotJuggler better. Therefore: thanks!

But the time I have is limited and I will implement a new feature only if:

1. It scratches a hitch that I have myself already.
2. It has a considerable impact on a large percentage of my users.
3. It is sponsored as paid development.

If you were redirected to this page from an issue marked as [consider_sponsorship](https://github.com/facontidavide/PlotJuggler/labels/consider%20sponsorship), then the development of this feature is **very** unlikely unless someone sponsors this work.

If you want to consider this option, please contact me at **davide.faconti@gmail.com**.

**Comment by franzhefekranz** [comment]
Hi, I'd like to have streaming from an updating csv-file. I'm recording from a PLC into a csv-file and would like to see that in "realtime."

**Reply by facontidavide** [reply]
Sure, let's discuss details by email. Davide.faconti@gmail.com

**Reply by Vadoola** [reply]
> Hi, I'd like to have streaming from an updating csv-file. I'm recording from a PLC into a csv-file and would like to see that in "realtime."

I'm curious what PLCs do you typically deal with?
I typically deal with Rockwell / Allen-Bradley was was considering on creating a PlotJuggler plugin using https://github.com/libplctag/libplctag to trend some things in real time. It's been a low priority and I just haven't had the time yet. Is this something other people would even be interested in?

**Comment by mickpeer** [comment]
Hello,
It would be great if multiple (two or more) Time TRackers could be added to Plot Window.
It would be useful e.g. in order to show on the plot a time between two plot points.

**Comment by Tmozix** [comment]
Hello, I would like PlotJuggler to support viewing historical modification parameters.

**Comment by vdemeter** [comment]
Hello, can you please decouple "Buffer" and plotted time window when plotting data from a stream? For example, I would like to have a buffer of 300sec but only show the latest 60s of data, and if I choose to pause, I want to be able to look at the data in the buffer by panning/zooming.

**Comment by jim-dp** [comment]
Hello,
Is it possible to add Linear Trend Removal in the filter list? The reason I am asking this one is that it seems it is not possible for the user to add it in Lua (de-trending methods like linear regression or least squares use loops on the the whole timeseries array, while the Lua user  functions in PlotJuggler can only work with one array value at a time).

**Comment by BenGundermannCnh** [comment]
I'd find it useful to be able to drag the cursor on a plot instead of relying on the slider on the bottom.  This is especially useful when zoomed into large logs.
